great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -484,6 +484,7 @@ def batch_parameters_to_batch_spec_kwarg_identifiers(
 ]
 
 
+@public_api
 class _SQLAsset(DataAsset[DatasourceT, ColumnPartitioner], Generic[DatasourceT]):
```

Comment:
Do we promote this one to `SQLAsset`?

Reply:
Similar to the pandas comment. Probably prepend Abstract to the name or leave as is. We want to communicate that it shouldn't be instantiated directly.

Reply:
I'd vote to rename in a follow-up if that's okay!